winrm/client.go

```diff
@@ -107,49 +114,84 @@ func (client *Client) NewShell(shellID string) *Shell {
 }
 
 func (client *Client) sendRequest(request *soap.SoapMessage) (response string, err error) {
+	if errorCode, err := client.check(); errorCode != 0 {
+		return "", err
+	}
 	return client.http(client, request)
 }
 
+// Use this function to map any know terminating errors to exitcodes and messages
+func (client *Client) check() (exitCode int, err error) {
+	return FindErrorFromServerError(client.err)
+}
+
+// Check if we need to return an ExitCode
+func (client *Client) ExitCode(cmd *Command) (exitCode int) {
+	if cmd != nil {
+		if cmd.ExitCode() != 0 {
+			return cmd.exitCode
+		}
+	}
+	exitCode, _ = client.check()
+	return exitCode
```

Comment:
If you give the return besides his type a name  it automatically **allocs on the stack** that *var* of that *type* (in our case **int**) so it's a little **redundant** to use ```return nameParam``` and you can write this as a **naked** return ``` return ``` but naked returns are *bad* and it's not *idiomatic* way to do this and it's a little difficult to review for other contribuitors the code especially for longer functions and I suggest to eliminate the return name var and just specify the type of return and alloc that var somewhere in the func semantic and return it like ``` return scopeVar ```.
A good plus for specifing a name on the return param is just for documentation purpose.
I hope you @masterzen  will decide what is better, A good point is for the future it will be very good to make a **.md** document with code guidelines this way we will eliminate inconsistency in the code and make a much more cleaner syntactically.